code/drv_can/src/can_sniffer/can_sniffer.py

```diff
@@ -103,59 +116,45 @@ def match(self, id_can: int) -> bool:
 
-class DrvCanParamsC:
-    """
-    Class that contains the can parameters in order to create the thread correctly
-    """
-    def __init__(self, target: Callable[..., object] | None = ...,
-        name: str | None = ..., args: Iterable[Any] = ...,
-        kwargs: Mapping[str, Any] | None = ..., *, daemon: bool | None = ...):
-        self.target = target
-        self.name = name
-        self.args = args
-        self.kwargs = kwargs
-        self.daemon = daemon
-
-class DrvCanNodeC(threading.Thread):
-    """Returns a removable version of the DRv command .
-
-    Args:
-        threading ([type]): [description]
+class DrvCanNodeC(SysShdNodeC): #pylint: disable= abstract-method
+    """Class to manage the CAN communication.
     """
 
-    def __init__(self, tx_buffer: SysShdChanC,
-        working_flag : threading.Event, can_params: DrvCanParamsC =
-        DrvCanParamsC()) -> None:
-        '''
-        Initialize the thread node used to received messages from CAN network.
+    def __init__(self, tx_buffer_size: int, working_flag : Event, name: str= "CAN_NODE",
+                cycle_period: int= 200,
+                can_params: SysShdNodeParamsC = SysShdNodeParamsC()) -> None:
+        """ Initialize the CAN node.
 
         Args:
-            chanPlak (SysShdChanC): Chan used to store messages received from plaks.
-            chanEPC (SysShdChanC): Chan used to store messages received from EPCs.
-        '''
+            tx_buffer_size (int): [description]
+            working_flag (Event): [description]
+            name (str, optional): [description]. Defaults to "CAN_NODE".
+            cycle_period (int, optional): [Period in miliseconds]. Defaults to 100.
+            can_params (SysShdNodeParamsC, optional): [description]. Defaults to SysShdNodeParamsC()
+        """
 
-        super().__init__(group = None, target = can_params.target, name = can_params.name,
-                         args = can_params.args, kwargs = can_params.kwargs,
-                         daemon = can_params.daemon)
+        super().__init__(name=name, cycle_period=cycle_period, working_flag=working_flag,
+                        node_params=can_params)
         self.working_flag = working_flag
         # cmd_can_down = 'sudo ip link set down can0'
         self.__can_bus : ThreadSafeBus = ThreadSafeBus(interface='socketcan',
                                 channel='can0', bitrate=125000,
                                 receive_own_messages=False, fd=True)
         self.__can_bus.flush_tx_buffer()
 
-        self.tx_buffer: SysShdChanC = tx_buffer
-
-        self.__active_filter = []
+        self.tx_buffer: SysShdIpcChanC = SysShdIpcChanC(name='TX_CAN',
+                                            max_msg = int(tx_buffer_size),
+                                            max_message_size= _Constants.MAX_MESSAGE_SIZE)
 
+        self.__active_filter: list = []
 
     def __parse_msg(self, message: DrvCanMessageC) -> None:
```

Comment:
Use lambda functions if it works slowly.

Reply:
Not needed